Tree.cs

```diff
@@ -12,27 +12,99 @@ interface ITapable
     {
         bool Tap();
     }
+    enum AgeCatagory
+    {
+        Young,
+        Mature,
+        Elderly,
+        Unknown
+    }
     class Tree
     {
+        //TODO: Make abstract
+        public Tree(int age_days)
+        {
+            Age_Days = age_days;
+        }
+        //AGE:
         protected int _age;
-        protected bool _diseased;
+        protected (int yearStart, AgeCatagory ageCatagory)[] _ageRanges; // Array of tuples :D
+
+        public int Age_Days
+        {
+            get { return _age; }
+            protected set
+            {
+                if (value >= 0) { _age = value; }
+                else {
+                    throw new ArgumentOutOfRangeException(String.Format("{0} is a negative age.", value), "value");
+                }
```

Comment:
@TimMcGilly idk if "value" is the right thing for here, but I think they just had the var which was out of range in the example I looked at, which I guess is value

Reply:
@tomkettle1 Arguments should be other way round if using two. https://docs.microsoft.com/en-us/dotnet/api/system.argumentoutofrangeexception.-ctor?view=net-5.0#System_ArgumentOutOfRangeException__ctor_System_String_System_String_

The "value" is the name of incorrect parameter so I would either change "value" to "Age_Days" or just remove it completely as I don't think it adds much value